### PR TITLE
Fix regression tests for cached Transactions

### PR DESCRIPTION
**Pull request by acha-bill**
# Description

Fix failed regression tests after caching `TransactionViewModels`.

I realized that when running `BundleValidator#loadTransactionsFromTangle` for a given tx, it could happen that the `tx.branch` or `tx.trunk` could be older than what's in the cache. So we read from cache always.

Secondly, fresh tx i.e `parsed=false` were being gossiped to the node and `tx.store` was overriding the already parsed tx in the cache. So some sort of `cache.putIfAbsent` was necessary for `tx.store`.
Only way cached tx update is by `tx.update`

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

# How Has This Been Tested?

- Existing unit tests pass

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/main/java/com/iota/iri/controllers/TransactionViewModel.java [hunk 1]
```diff
@@ -2,32 +2,16 @@
...
-import com.iota.iri.model.TagHash;
-import com.iota.iri.model.TransactionHash;
-import com.iota.iri.model.persistables.Address;
-import com.iota.iri.model.persistables.Approvee;
-import com.iota.iri.model.persistables.Bundle;
-import com.iota.iri.model.persistables.ObsoleteTag;
-import com.iota.iri.model.persistables.Tag;
-import com.iota.iri.model.persistables.Transaction;
+import com.iota.iri.model.*;
+import com.iota.iri.model.persistables.*;
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/controllers/TransactionViewModel.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.model.persistables'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4795345)

File: src/main/java/com/iota/iri/controllers/TransactionViewModel.java [hunk 2]
```diff
@@ -2,32 +2,16 @@
...
-import com.iota.iri.model.ObsoleteTagHash;
-import com.iota.iri.model.TagHash;
-import com.iota.iri.model.TransactionHash;
-import com.iota.iri.model.persistables.Address;
-import com.iota.iri.model.persistables.Approvee;
-import com.iota.iri.model.persistables.Bundle;
-import com.iota.iri.model.persistables.ObsoleteTag;
-import com.iota.iri.model.persistables.Tag;
-import com.iota.iri.model.persistables.Transaction;
+import com.iota.iri.model.*;
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/controllers/TransactionViewModel.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.model'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4795345)

File: python-regression/util/test_logic/api_test_logic.py [hunk 3]
```diff
@@ -133,7 +134,8 @@ def fetch_call(api_call, api, options):
 
     try:
         response = call_list[api_call](**options)
-    except ValueError:
+    except ValueError, e:
+        logger.info(str(e))
```
[the remark this answers is not in the document]
Comment (acha-bill) [reply python-regression/util/test_logic/api_test_logic.py]: sure.

File: src/main/java/com/iota/iri/controllers/TransactionViewModel.java (hunk 1 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/controllers/TransactionViewModel.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.model.persistables'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4795345)

File: src/main/java/com/iota/iri/controllers/TransactionViewModel.java (hunk 2 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/controllers/TransactionViewModel.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.model'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4795345)

**Pull request merged**